### PR TITLE
refactor: refer to the entry point instead of the first `module` (`module.identifier`)

### PR DESCRIPTION
**Pull request by akihikodaki**
The order of modules can be reordered by `optimize-module-order` plugins. The intention is to use the entry point as the default identifier, so refer to entries property of the compilation.

### CODE REVIEW

**Comment by akihikodaki** [comment]
Commit 426e758b931fb67d711e25bcf57a5a5ad1c88102 adds a test case.

**Comment by akihikodaki** [comment]
See also https://github.com/webpack/webpack.js.org/pull/1527.

**Comment by akihikodaki** [comment]
https://github.com/miraiken/web/commit/46dfdde7694f232115b1d57bea7c0505cca13194 is an easy workaround.

**Comment by evilebottnawi** [comment]
@akihikodaki thanks for PR

**Comment by evilebottnawi** [comment]
@sokra need review

File: src/loader.js [hunk 1]
```diff
@@ -94,13 +94,16 @@ export function pitch(request) {
       }
       try {
         let text = this.exec(source, request);
-        if (typeof text === 'string') { text = [[0, text]]; }
-        text.forEach((item) => {
-          const id = item[0];
-          compilation.modules.forEach((module) => {
-            if (module.id === id) { item[0] = module.identifier(); }
+        if (typeof text === 'string') {
+          text = [[compilation.entries[0].identifier(), text]];
```
Comment (sokra) [inline src/loader.js]: You can use `[[entries[0].identifier(), text]]` too i guess.

**Comment by evilebottnawi** [comment]
@akihikodaki friendly ping

**Comment by michael-ciniawsky** [comment]
### `Issues`

- Fixes #648

**Comment by sgal** [comment]
Also fixes https://github.com/webpack/webpack/issues/5733

**Comment by akihikodaki** [comment]
52a194284a5dfd4c8a1543c82c0cd773ee447978 applies https://github.com/webpack-contrib/extract-text-webpack-plugin/pull/601#pullrequestreview-70527320.

**Comment by d3viant0ne** [comment]
@akihikodaki - Once #652 lands, this will need one more quick rebase before it merges.

**Comment by michael-ciniawsky** [comment]
@akihikodaki #652 just landed please rebase when time :)

**Comment by akihikodaki** [comment]
8228819873bba36aded831f5b72e06189ac8c35a: rebased to 5286ab20ad83f6ade30192f939e0b42d3956d6f5.

**Comment by michael-ciniawsky** [comment]
> Module build failed: TypeError: entries[0].identifier is not a function

@akihikodaki Can you revert to `compilation.entries[0].identifier` ?  Or create a new variable etc ?

**Comment by akihikodaki** [comment]
f2c3056c25e35e807b74fa208ca50899c0af6708: Reverted the change. I reviewed the intension why I used `compilation.entries`, and it was actually because `entries` argument is not an array of the first modules to be executed, but of chunks.

**Comment by michael-ciniawsky** [comment]
https://github.com/webpack-contrib/extract-text-webpack-plugin/blob/f2c3056c25e35e807b74fa208ca50899c0af6708/src/loader.js#L81 did you add this 🙃 , it seems to be unused ?

**Comment by akihikodaki** [comment]
No, I didn't.

**Comment by cletusw** [comment]
Thanks! Could someone post here when we get a release bump with this fix?

**Comment by sgal** [comment]
Is there any plan to release a new version with this fix? The issue with `NamedModulesPlugin` and `HashedModuleIdsPlugin` is preventing users from implementing [long term caching solution](https://webpack.js.org/guides/caching/), so it would be nice to get it out.

**Pull request merged**